uniprotpy/parser.py

```diff
@@ -0,0 +1,35 @@
+import re
+from Bio import SeqIO
+
+def parse_protein_string(s):
+    pattern = r'^tr\|(\w+)\|(\w+)_\w+\s+(.+) OS=(.+) OX=(\d+) GN=(\w+) PE=(\d+) SV=(\d+)(?: GP=(\d+))?$'
```

Comment:
Does this pattern handle if the piece of data is not there?

Sometimes in the FASTA header, the "GN=" or the "SV=" is missing, which is why I used separate regexes to pull them out. Maybe we should keep it that way? We should write tests that test a few different proteomes from UniProt so that we can catch these things.

Reply:
The code can be modified for missing data as seen at the bottom. But yes I agree, the tests should be more comprehensive. I only modified the regular expressions because I hate regular expressions (chatGPT helped me write this one I won't lie). I was trying to work with lookahead and lookbehind and struggling 😭 

```
    if match:
        return {
            "protein_id": match.group(2),
            "protein_name": match.group(3),
            "species": match.group(4),
            "taxon_id": match.group(5),
            "gene": match.group(6),
            "pe_level": match.group(7),
            "sequence_version": match.group(8) if match.group(8) else "",
            "gene_priority": match.group(9) if match.group(9) else "",
        }
        ```

Reply:
Haha, yeah, I have tested if a few things, using some proteomes, but haven't written any actual tests. Sometimes, all the header data is missing except for the ID. 

Let's change it back to the original way and then we can merge the PR.